spring-webflux/src/main/java/org/springframework/web/reactive/socket/adapter/JettyWebSocketSession.java

```diff
@@ -108,25 +175,81 @@ public boolean isOpen() {
 	public Mono<Void> close(CloseStatus status) {
 		Callback.Completable callback = new Callback.Completable();
 		getDelegate().close(status.getCode(), status.getReason(), callback);
-
 		return Mono.fromFuture(callback);
 	}
 
+	@Override
+	public Mono<CloseStatus> closeStatus() {
+		return this.closeStatusSink.asMono();
+	}
 
-	private final class SendProcessorCallback implements Callback {
+	@Override
+	public Flux<WebSocketMessage> receive() {
+		return this.flux;
+	}
 
-		@Override
-		public void fail(Throwable x) {
-			getSendProcessor().cancel();
-			getSendProcessor().onError(x);
-		}
+	@Override
+	public Mono<Void> send(Publisher<WebSocketMessage> messages) {
+		return Flux.from(messages)
+				.flatMap(this::sendMessage, 1)
+				.then();
+	}
 
-		@Override
-		public void succeed() {
-			getSendProcessor().setReadyToSend(true);
-			getSendProcessor().onWritePossible();
-		}
+	protected Mono<Void> sendMessage(WebSocketMessage message) {
 
+		Callback.Completable completable = new Callback.Completable();
+		DataBuffer dataBuffer = message.getPayload();
+		Session session = getDelegate();
+		if (WebSocketMessage.Type.TEXT.equals(message.getType())) {
+			String text = dataBuffer.toString(StandardCharsets.UTF_8);
+			session.sendText(text, completable);
+		}
+		else {
+			switch (message.getType()) {
+				case BINARY -> {
+					@SuppressWarnings("resource")
+					DataBuffer.ByteBufferIterator iterator = dataBuffer.readableByteBuffers();
+					new IteratingCallback() {
+						@Override
+						protected Action process() {
+							if (!iterator.hasNext()) {
+								return Action.SUCCEEDED;
+							}
+
+							ByteBuffer buffer = iterator.next();
+							boolean last = iterator.hasNext();
+							session.sendPartialBinary(buffer, last, Callback.from(this::succeeded, this::failed));
```

Comment:
```suggestion
							session.sendPartialBinary(buffer, last, this);
```

Reply:
Can't do, one is the WebSocket API callback one is the jetty.util callback.